pkg/controller/operators/olm/operator.go

```diff
@@ -296,6 +301,50 @@ func NewOperator(logger *logrus.Logger, crClient versioned.Interface, opClient o
 	return op, nil
 }
 
+func (a *Operator) syncAPIService(obj interface{}) (syncError error) {
+	apiSvc, ok := obj.(*apiregistrationv1.APIService)
+	if !ok {
+		a.Log.Debugf("wrong type: %#v", obj)
+		return fmt.Errorf("casting APIService failed")
+	}
+
+	logger := a.Log.WithFields(logrus.Fields{
+		"id":     queueinformer.NewLoopID(),
+		"apiSvc": apiSvc.GetName(),
+	})
+	logger.Info("syncing APIService")
+
+	serviceLabels := apiSvc.GetLabels()
+	owningNamespace, owned := serviceLabels[ownerutil.OwnerNamespaceKey]
```

Comment:
we should check if labels are nil before accessing

Reply:
Refactored to use GetOwnerByKindLabel.